src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorEditorFactory.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Microsoft.Internal.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Package;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Guid(EditorFactoryGuidString)]
+    public class RazorEditorFactory : EditorFactory
+    {
+        private const string EditorFactoryGuidString = "3dfdce9e-1799-4372-8aa6-d8e65182fdfc";
+        private const string RazorLSPEditorFeatureFlag = "Razor.LSP.Editor";
+        private readonly Lazy<IVsEditorAdaptersFactoryService> _adaptersFactory;
+        private readonly Lazy<IContentType> _razorLSPContentType;
+        private readonly Lazy<IVsFeatureFlags> _featureFlags;
+
+        public RazorEditorFactory(AsyncPackage package) : base(package)
+        {
+            _adaptersFactory = new Lazy<IVsEditorAdaptersFactoryService>(() =>
+            {
+                var componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
+                var adaptersFactory = componentModel.GetService<IVsEditorAdaptersFactoryService>();
+                return adaptersFactory;
+            });
+
+            _razorLSPContentType = new Lazy<IContentType>(() =>
+            {
+                var componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
+                var contentTypeService = componentModel.GetService<IContentTypeRegistryService>();
+                var contentType = contentTypeService.GetContentType(RazorLSPContentTypeDefinition.Name);
+                return contentType;
+            });
+
+            _featureFlags = new Lazy<IVsFeatureFlags>(() =>
+            {
+                var featureFlags = (IVsFeatureFlags)AsyncPackage.GetGlobalService(typeof(SVsFeatureFlags));
+                return featureFlags;
+            });
+        }
+
+        private bool IsLSPRazorEditorEnabled
```

Comment:
nit: in other places you have RazorLSP and this is LSPRazor. I think the first is easier to read (capital P and R and not happy).